packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -13,40 +14,99 @@ export const useTypeBrowser = (onDoubleClick: () => void): UseBrowserImplReturnV
     id: TYPE_BROWSER_ID,
     name: 'Type',
     content: <TypeBrowser value={value} onChange={setValue} onDoubleClick={onDoubleClick} />,
-    accept: () => value
+    accept: () => value,
+    icon: IvyIcons.DataClass
   };
 };
 
+type TypeKind = 'DATACLASS' | 'NORMALTYPE' | 'OWNTYPE';

```

Comment:
normally I would say this kind is perfect. But as far as I understand you code, you currently only need this to differ the icons right?
If this is the case I think this is a little "overkill" and more or less useless code which needs to run. Better just add directly the correct icon to the TypeBrowserObject.